File: gradle-plugin/src/main/java/com/kategory/ank/plugin/AnkGradlePlugin.java
package com.kategory.ank.plugin;


import org.gradle.api.Plugin;
import org.gradle.api.Project;
import org.gradle.api.tasks.SourceSetContainer;

public class AnkGradlePlugin implements Plugin<Project> {
  static final String TASK_NAME = "ank";

  @Override
  public void apply(Project target) {
    SourceSetContainer sourceSets = (SourceSetContainer)
      target.getProperties().get("sourceSets");
    sourceSets.getByName("main").getRuntimeClasspath();
    target.getTasks().create(TASK_NAME, AnkTask.class);
  }
}

File: gradle-plugin/plugin/src/main/kotlin/com/kategory/ank/plugin/AnkGradlePlugin.kt
package com.kategory.ank.plugin

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.Task
import org.gradle.api.tasks.Exec

import org.gradle.kotlin.dsl.*

open class AnkGradlePlugin : Plugin<Project> {

    override fun apply(project: Project) {
        project.run {

            tasks {
                "myEchoTask"(Exec::class) {
                    executable("sh")
                    args("-c", "echo 'Hello ΛNK from your shell!'")
                }
            }
        }
    }
}
